apps/demos/Demos/Chat/AIIntagration/jQuery/index.js

```diff
@@ -0,0 +1,189 @@
+const deployment = 'gpt-4o-mini';
+const apiVersion = '2024-02-01';
+const endpoint = 'https://public-api.devexpress.com/demo-openai';
+const apiKey = 'DEMO';
+
+const store = [];
+let messages = [];
+let lastAIResponse = null;
+
+const user = {
+  id: 'user',
+};
+
+const assistant = {
+  id: 'assistant',
+  name: 'Virtual Assistant',
+};
+
+$(() => {
+  DevExpress.localization.loadMessages({
+    "en": {
+      "dxChat-emptyListMessage": "Virtual Assistant",
+      "dxChat-emptyListPrompt": "An AI bot ready to provide instant support for your needs.",
+    },
+  });
+
+  const chatService = new AzureOpenAI({
+    dangerouslyAllowBrowser: true,
+    deployment,
+    endpoint,
+    apiVersion,
+    apiKey,
+  });
+
+  async function getAIResponse(messages) {
+    const params = {
+      messages: messages,
+      max_tokens: 1000,
+      temperature: 0.7,
+    };
+
+    const responseAzure = await chatService.chat.completions.create(params);
+    const data = { choices: responseAzure.choices };
+
+    return data.choices[0].message?.content;
+  };
+
+  async function processMessageSending() {
+    instance.option({ typingUsers: [assistant] });
+
+    try {
+      const aiResponse = await getAIResponse(messages);
+
+      setTimeout(() => {
+        instance.option({ typingUsers: [] });
+
+        messages.push({ role: 'assistant', content: aiResponse });
+
+        renderMessage(aiResponse);
+      }, 200);
+    } catch {
+      instance.option({
+        typingUsers: [],
+        alerts: [{ message: 'Request limit reached, try again in a minute.' }],
+      });
+
+      if (lastAIResponse) {
+        messages.push(lastAIResponse);
+        renderMessage(lastAIResponse.content);
+        lastAIResponse = null;
+      }
+
+      setTimeout(() => {
+        instance.option({ alerts: [] });
+      }, 10000);
+    }
+  };
+
+  function renderMessage(text) {
+    const message = {
+      id: Date.now(),
+      timestamp: new Date(),
+      author: assistant,
+      text,
+    };
+
+    updateStore('insert', message);
+  };
+
+  function updateStore(type, data, key) {
+    customStore.push([{ type, data, key }]);
+  };
+
+  function prepareDataToRefresh() {
+    const { items } = instance.option();
+    const lastMessage = items[items.length - 1];
```

Comment:
I suggest you using this approach everywhere
```suggestion
    const lastMessage = items.at(-1);
```

Reply:
[0f60119da81def3f3f4e8a295eaefda87674892c]